test/create-history-router.test.ts

```diff
@@ -129,6 +129,145 @@ describe("Initialization", () => {
       query: {},
     });
   });
+
+  it("Triggers route.opened on .setHistory() call (hydrate is not set)", async () => {
+    const history = createMemoryHistory({
+      initialEntries: ["/foo"],
+    });
+
+    const opened = watch(foo.opened);
+    const scope = fork();
+
+    expect(opened).toBeCalledTimes(0);
+
+    await allSettled(router.setHistory, {
+      scope,
+      params: history,
+    });
+
+    expect(opened).toBeCalledTimes(1);
+  });
+
+  it("Triggers route.opened on .setHistory() call (hydrate: false)", async () => {
+    const hydratedRouter = createHistoryRouter({
+      routes: [
+        { route: foo, path: "/foo" },
+        { route: bar, path: "/bar" },
+        { route: first, path: "/first" },
+        { route: firstClone, path: "/first" },
+        { route: withParams, path: "/posts/:postId" },
+        { route: hashed, path: "/test/#/swap/:token" },
+      ],
+      controls,
+      /**
+       * Explicitly set hydrate to true
+       */
+      hydrate: false,
+    });
```

Comment:
Explicitly set hydrate to true, but hydrate false

Reply:
Fixed ✅